portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,18 +42,56 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    ArrayList<String> messages = new ArrayList<>();
+    HashMap<Integer, String> cursors = new HashMap<Integer, String>();
```

Comment:
NIT: Keep the type declaration of the variable to be general (i.e. Map instead of HashMap.)
Consider instantiating cursors, pageCount on the first doGet run.